src/main/java/christmas/domain/BenefitDetails.java

```diff
@@ -0,0 +1,117 @@
+package christmas.domain;
+
+import christmas.domain.event.badge.BadgeEvent;
+import christmas.domain.event.discount.DDayEvent;
+import christmas.domain.event.discount.DiscountEvent;
+import christmas.domain.event.discount.SpecialEvent;
+import christmas.domain.event.discount.WeekdayEvent;
+import christmas.domain.event.discount.WeekendEvent;
+import christmas.domain.event.giveaway.ChampagneEvent;
+import christmas.domain.event.giveaway.Giveaway;
+import christmas.domain.event.giveaway.GiveawayEvent;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BenefitDetails {
+
+    private static final int PARTICIPATION_CRITERIA = 10000;
+    private static final int MIN_AMOUNT = 0;
+    public static final int EVENT_START_DAY = 1;
+    public static final int EVENT_LAST_DAY = 31;
+    private final List<String> discountNames;
+    private final List<Integer> discountAmount;
+    private int totalDiscountAmount;
+    private String badge;
+    private Giveaway giveaway;
+
+    public BenefitDetails() {
+        this.discountNames = new ArrayList<>();
+        this.discountAmount = new ArrayList<>();
+    }
+
+    public static BenefitDetails getBenefitDetails() {
+        return new BenefitDetails();
+    }
+
+    public void receive(int date, Bill bill) {
+        if (bill.getTotalAmount() >= PARTICIPATION_CRITERIA &&
+                date >= EVENT_START_DAY && date <= EVENT_LAST_DAY) {
+            getDiscountInfo(date, bill);
+        }
+        getGiveawayInfo(bill);
+        this.totalDiscountAmount = discountAmount.stream().mapToInt(Integer::intValue).sum();
```

Comment:
코드 컨벤션을 지켜서 해당 방식으로 바꿔주시면 좋으실거 같아요~!
```suggestion
        this.totalDiscountAmount = discountAmount.stream()
        .mapToInt(Integer::intValue)
        .sum();
```